doc/design/dynamic_tls.md

```diff
@@ -0,0 +1,122 @@
+# Dynamic TLS Certs
+
+Authors: Eric Chiang(eric.chiang@coreos.com), Hongchao Deng(hongchao.deng@coreos.com)
+
+## Background
+
+Motivations:
+- Simplify TLS cert generation for etcd cluster.
+- Self-hosted etcd wants to depend on Node IP, not hostnames.
+
+Requirements:
+- Dynamically generate certs per etcd pod
+- Able to use host IP in self hosted.
+
+
+## Proposed solution
+
+### Design
+
+> For initial design, we focus on self-hosted etcd use case, although the same methodology applies to app etcd.
+
+[Kubernetes certificates API][k8s_csr] provides the ability to send and approve CSR.
```

Comment:
This case was covered in the bootkube auto-TLS proposal. Might want to link to that since it talks about this case a little more in depth?

https://docs.google.com/document/d/1POXVGyEoySvSnx_OftQ2CIWM0HCk27j2VZSOR4XVCDg/edit#heading=h.o7y0uvuoaz4a